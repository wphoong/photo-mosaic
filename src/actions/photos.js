import uuid from "uuid";

// ADD PHOTO

export const addPhoto = (
  {
    photoLink = "",
    title = "",
    createdAt = 0,
    description = ""
  } = {}) => ({
    type: "ADD_PHOTO",
    photo: {
      id: uuid(),
      photoLink,
      title,
      createdAt,
      description
    }
  }
);

export const editPhoto = (id, updates) => ({
  type: "EDIT_PHOTO",
  id,
  updates
});

export const removePhoto = ({id} = {}) => ({
  type: "REMOVE_PHOTO",
  id
});

// export const startAddPhoto = (photoData = {}) => {
//   return (dispatch, getState) => {
//     const uid = getState().auth.uid;
//     const {
//       photoLink = "",
//       title = "",
//       description = "",
//       createdAt = 0
//     } = photoData;
//     const photo = { photoLink, title, description, createdAt };
    
//     return dispatch(addPhoto({
//       id: ref.key,
//       ...photo 
//     }));
//   };
// };