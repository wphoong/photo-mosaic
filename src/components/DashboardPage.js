import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import 'react-dates/initialize';
import PhotoListFilters from "./PhotoListFilters.js";
import PhotoList from "./PhotoList.js";
import { firebase } from "../firebase/firebase.js";
import getVisiblePhotos from "../selectors/photos.js";

let userName;

firebase.auth().onAuthStateChanged((user) => {
  if (user) {
    console.log("name", user.displayName);
    userName = user.displayName;
  } else {
    console.log("no user signed in");
  }
});

export const DashboardPage = (props) => (
  <div className="content-container">
    <h1 className="header__secondary title is-2">{userName}'s Photos</h1>
    { props.photos.length !== 0 && <h3 className="header__secondary">Showing {props.photos.length} Photo(s)</h3>}
    <Link className="button is-primary is-outlined button--photo" to="/create">
      <h2 >Add Photo</h2>
    </Link>
    <PhotoListFilters />
    <PhotoList />
  </div>
);

const mapStateToProps = (state) => ({
  photos: getVisiblePhotos(state.photos, state.filters)
});

export default connect(mapStateToProps)(DashboardPage);

