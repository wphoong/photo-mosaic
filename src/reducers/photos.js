const photosReducerDefaultState = [];

const photosReducer = (state = photosReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_PHOTO":
      return [
        ...state,
        action.photo
      ];
    default:
      return state;
  }
};

export default photosReducer;