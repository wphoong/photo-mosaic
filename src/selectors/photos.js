import moment from "moment";

const getVisiblePhotos = (photos, { text, sortBy, startDate, endDate}) => {
  return photos.filter((photo) => {
    const createdAtMoment = moment(photos.createdAt);
    const setTextMatch = photo.title.toLowerCase().includes(text.toLowerCase());
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, "day") : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, "day") : true;

    return setTextMatch && startDateMatch && endDateMatch;
  }).sort((a, b) => {
      if (sortBy === "descend") {
        return b.createdAt - a.createdAt;
      } else {
        return a.createdAt - b.createdAt;
      }
    }
  );
};

export default getVisiblePhotos;
