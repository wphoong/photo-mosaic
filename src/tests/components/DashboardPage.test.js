import React from "react";
import { shallow } from "enzyme";
import { DashboardPage } from "../../components/DashboardPage.js";
import photos from "../fixtures/photos.js";

test("should render expense dashboard page correctly", ()=> {
  const wrapper = shallow(<DashboardPage photos={photos} />);
  expect(wrapper).toMatchSnapshot();
});