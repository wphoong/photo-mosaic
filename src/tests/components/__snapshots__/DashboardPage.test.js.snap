// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense dashboard page correctly 1`] = `
<div
  className="content-container"
>
  <h1
    className="header__secondary title is-2"
  >
    's Photos
  </h1>
  <h3
    className="header__secondary"
  >
    Showing 
    3
     Photo(s)
  </h3>
  <Link
    className="button is-primary is-outlined button--photo"
    replace={false}
    to="/create"
  >
    <h2>
      Add Photo
    </h2>
  </Link>
  <Connect(PhotoListFilters) />
  <Connect(PhotoList) />
</div>
`;
